### PR TITLE
MueLu: UncoupledAggregation_kokkos fixes

### PR DESCRIPTION
**Pull request by cgcgcg**
@trilinos/muelu 

## Motivation
- UncoupledAggregation_kokkos was ignoring Dirichlet nodes.
- TentativeP_kokkos did not compress the matrix correctly for nullspace of dim = 1.

### CODE REVIEW

File: packages/muelu/src/Graph/UncoupledAggregation/MueLu_UncoupledAggregationFactory_kokkos_def.hpp [hunk 1]
```diff
@@ -196,14 +196,19 @@ namespace MueLu {
...
     // instead of an ArrayRCP.
-    // // TODO
-    // //ArrayRCP<const bool> dirichletBoundaryMap = graph->GetBoundaryNodeMap();
-    // ArrayRCP<const bool> dirichletBoundaryMap;
-
-    // if (dirichletBoundaryMap != Teuchos::null)
-    //   for (LO i = 0; i < numRows; i++)
-    //     if (dirichletBoundaryMap[i] == true)
-    //       aggStat[i] = BOUNDARY;
+    {
```
Comment (lucbv) [inline packages/muelu/src/Graph/UncoupledAggregation/MueLu_UncoupledAggregationFactory_kokkos_def.hpp]: Added issue #6269 so we don't forget to improve this bit of code later

**Comment by rppawlo** [comment]
@cgcgcg looks like this has been approved and passed PR testing. Did you want this merged?

**Comment by jhux2** [comment]
@cgcgcg Did you by any chance see if this affected the Kokkos-path negative # of aggregates problem?

**Comment by cgcgcg** [comment]
Not yet. I'll have to pull and rebuild on lassen..

**Comment by cgcgcg** [comment]
@jhux2 Nope, not fixed.

**Pull request merged**